llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -2534,6 +2535,20 @@ void SITargetLowering::allocatePreloadKernArgSGPRs(
       unsigned NumAllocSGPRs =
           alignTo(ArgLoc.getLocVT().getFixedSizeInBits(), 32) / 32;
 
+      // Add padding SPGR to fix alignment for hidden arguments.
+      if (!AlignedForImplictArgs && Arg.hasAttribute("amdgpu-work-group-id")) {
```

Comment:
There is no such attribute as "amdgpu-work-group-id", all of this is dead?

Reply:
Thanks for catching this, it must have become dead at some point in the review process and the tests autoupdated without noticing.